lib/resources/forms.js

```diff
@@ -136,6 +136,18 @@ module.exports = (service, endpoint) => {
       .then(((form) => Promise.all([ Forms.publish(form), Submissions.clearDraftSubmissions(form.id) ])))
       .then(success)));
 
+  // Entity/Dataset-specific endpoint that is used to show how publishing
+  // a form will change any datasets mentioned in the form.
+  // Even though there are dataset-related permissions, we will use the form modification verbs
+  // for authentication because it seems weird to mix the ability/inability to alter a dataset
+  // with the ability to modify forms -- they should just be the same.
```

Comment:
Above, I suggest using a different verb than `form.update`, but I agree with the thoughts in this comment. It seems like updating a form and updating a dataset go hand-in-hand.